src/claim_canister/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+    });
+
+    Ok(claim_id)
+}
+
+#[ic_cdk::update]
+pub fn approve_claim(claim_id: u64) -> Result<(), ClaimError> {
+    let caller = ic_cdk::api::caller();
+
+    let is_approver = APPROVERS.with(|approvers| approvers.borrow().get(&caller).unwrap_or(false));
+
+    if !is_approver {
+        return Err(ClaimError::NotApprover);
+    }
+
+    CLAIMS.with(|claims| {
+        let mut claims_ref = claims.borrow_mut();
+        let mut claim = claims_ref.get(&claim_id).ok_or(ClaimError::NotFound)?;
+
+        if claim.status == ClaimStatus::Pending {
+            claim.status = ClaimStatus::Approved;
+            claim.approved_at = Some(ic_cdk::api::time());
+            claim.approved_by = Some(caller);
+            claims_ref.insert(claim_id, claim);
+            Ok(())
+        } else if claim.status == ClaimStatus::Approved {
+            Err(ClaimError::AlreadyApproved)
+        } else if claim.status == ClaimStatus::Executed {
+            Err(ClaimError::AlreadyExecuted)
+        } else {
+            Err(ClaimError::InvalidStatus)
+        }
+    })
+}
+
+#[ic_cdk::update]
+pub async fn execute_claim(claim_id: u64) -> Result<(), ClaimError> {
+    let (pool_canister_id, receiver, amount) = CLAIMS.with(
+        |claims| -> Result<(Principal, Principal, Nat), ClaimError> {
+            let mut claims_ref = claims.borrow_mut();
+            let mut claim = claims_ref.get(&claim_id).ok_or(ClaimError::NotFound)?;
+
+            if claim.status == ClaimStatus::Executed {
+                return Err(ClaimError::AlreadyExecuted);
+            }
+            if claim.status != ClaimStatus::Approved {
+                return Err(ClaimError::NotApproved);
+            }
+
+            let current_time = ic_cdk::api::time();
+            let approved_time = claim.approved_at.ok_or(ClaimError::NotApproved)?;
+
+            if current_time < approved_time + TIMELOCK_DURATION {
+                return Err(ClaimError::TimelockNotExpired);
+            }
+
+            claim.status = ClaimStatus::Executing;
+            let pool_canister_id = claim.pool_canister_id;
+            let receiver = claim.receiver;
+            let amount = claim.amount.clone();
+
+            claims_ref.insert(claim_id, claim);
+            Ok((pool_canister_id, receiver, amount))
+        },
+    )?;
+
+    let slash_result: Result<(Result<(), PoolError>,), _> =
+        call(pool_canister_id, "slash", (receiver, amount)).await;
+
+    let success = matches!(slash_result, Ok((Ok(()),)));
+
+    if !success {
+        CLAIMS.with(|claims| {
+            let mut claims_ref = claims.borrow_mut();
+            if let Some(mut updated_claim) = claims_ref.get(&claim_id) {
+                updated_claim.status = ClaimStatus::Approved;
+                claims_ref.insert(claim_id, updated_claim);
+            }
+        });
+
+        return Err(ClaimError::PoolCallFailed(format!(
+            "{:?}",
+            slash_result.unwrap().0.unwrap_err()
```

Comment:
I afraid there is a possible situation when we may have not usual Ok(Err()) situation, but rather Err() on a first wrap. For example, when a pool canister doesn't have enough cycles, the whole call will faill.
So, format the whole error without unwrapping.